File: 3047-find-the-largest-area-of-square-inside-two-rectangles/3047-find-the-largest-area-of-square-inside-two-rectangles.py
class Solution:
    def largestSquareArea (self, bottomLeft : List [List [int]], topRight : List [List [int]]) -> int:
        maxSide = 0
        n = len (bottomLeft)
        for i in range (n):
            for j in range (i + 1, n):
                x1 = max (bottomLeft [i] [0], bottomLeft [j] [0])
                y1 = max (bottomLeft [i] [1], bottomLeft [j] [1])
                x2 = min (topRight [i] [0], topRight [j] [0])
                y2 = min (topRight [i] [1], topRight [j] [1])
                width = x2 - x1
                height = y2 - y1
                if width > 0 and height > 0:
                    side = min (width, height)
                    maxSide = max (maxSide, side)
        return maxSide * maxSide

File: 3047-find-the-largest-area-of-square-inside-two-rectangles/3047-find-the-largest-area-of-square-inside-two-rectangles.c
long long largestSquareArea (int** bottomLeft, int bottomLeftSize, int* bottomLeftColSize, int** topRight, int topRightSize, int* topRightColSize) {
    long long maxSide = 0;
    int n = bottomLeftSize;
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            int x1 = fmax (bottomLeft [i] [0], bottomLeft [j] [0]);
            int y1 = fmax (bottomLeft [i] [1], bottomLeft [j] [1]);
            int x2 = fmin (topRight [i] [0], topRight [j] [0]);
            int y2 = fmin (topRight [i] [1], topRight [j] [1]);
            long width = x2 - x1;
            long height = y2 - y1;
            if (width > 0 && height > 0) {
                long side = fmin (width, height);
                maxSide = fmax (maxSide, side);
            }
        }
    }
    return maxSide * maxSide;
}